lib/PuppeteerSharp.Tests/Page/CookiesTests.cs

```diff
@@ -0,0 +1,217 @@
+            Assert.Equal("Protocol error (Network.deleteCookies): At least one of the url and domain needs to be specified ", exception.Message);
+        }
+
+        [Fact]
+        public async Task ShouldNotSetACookieWithBlankPageURL()
+        {
+            var page = await Browser.NewPageAsync();
+            await page.GoToAsync(TestConstants.ServerUrl + "/grid.html");
+
+            var exception = await Assert.ThrowsAnyAsync<Exception>(async () => await page.SetCookieAsync(new CookieParam
+            {
+                Name = "example-cookie",
+                Value = "best"
+            }, new CookieParam
+            {
+                Url = TestConstants.AboutBlank,
+                Name = "example-cookie-blank",
+                Value = "best"
+            }));
+            Assert.Equal("Blank page can not have cookie \"example-cookie-blank\"", exception.Message);
+        }
+
+        [Fact]
+        public async Task ShouldNotSetACookieOnADataURLPage()
+        {
+            var page = await Browser.NewPageAsync();
+            await page.GoToAsync("data:,Hello%2C%20World!");
+            var exception = await Assert.ThrowsAnyAsync<Exception>(async () => await page.SetCookieAsync(new CookieParam { Name = "example-cookie", Value = "best" }));
+
+            Assert.Equal("Protocol error (Network.deleteCookies): At least one of the url and domain needs to be specified ", exception.Message);
+        }
+
+        [Fact]
+        public async Task ShouldSetACookieOnADifferentDomain()
+        {
+            var page = await Browser.NewPageAsync();
+            await page.GoToAsync(TestConstants.ServerUrl + "/grid.html");
+            await page.SetCookieAsync(new CookieParam { Name = "example-cookie", Value = "best", Url = "https://www.example.com" });
+            Assert.Equal(string.Empty, await page.EvaluateExpressionAsync<string>("document.cookie"));
+            Assert.Empty(await page.GetCookiesAsync());
+            var cookie = Assert.Single(await page.GetCookiesAsync("https://www.example.com"));
+            Assert.Equal(cookie.Name, "example-cookie");
+            Assert.Equal(cookie.Value, "best");
+            Assert.Equal(cookie.Domain, "www.example.com");
+            Assert.Equal(cookie.Path, "/");
+            Assert.Equal(cookie.Expires, -1);
+            Assert.Equal(cookie.Size, 18);
+            Assert.Equal(cookie.HttpOnly, false);
+            Assert.Equal(cookie.Secure, true);
+            Assert.Equal(cookie.Session, true);
+        }
+
+        [Fact(Skip = "Line182 doesn't return")]
+        public async Task ShouldSetCookiesFromAFrame()
+        {
+            var page = await Browser.NewPageAsync();
+            await page.GoToAsync(TestConstants.ServerUrl + "/grid.html");
+            await page.SetCookieAsync(new CookieParam { Name = "localhost-cookie", Value = "best" });
+            await page.EvaluateFunctionAsync(@"src => {
```

Comment:
method does not return from server

Reply:
Let me know if you need a hand man

Reply:
I think it's the first use of `EvaluateFunctionAsync` with arguments

Reply:
This test is using arguments https://github.com/kblok/puppeteer-sharp/blob/master/lib/PuppeteerSharp.Tests/Page/EvaluateTests.cs#L67

Reply:
I have no idea why this hangs :(

Reply:
I checked out this PR, pulled from master, and I got a green test!